internal/credential/static/kv_credential.go

```diff
@@ -0,0 +1,138 @@
+package static
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/hashicorp/boundary/internal/credential"
+	"github.com/hashicorp/boundary/internal/credential/static/store"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/libs/crypto"
+	"github.com/hashicorp/boundary/internal/oplog"
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	"github.com/hashicorp/go-kms-wrapping/v2/extras/structwrapping"
+	"google.golang.org/protobuf/proto"
+)
+
+var _ credential.Static = (*KvCredential)(nil)
+
+// A KvCredential contains the credential with a key-value secret.
+// It is owned by a credential store.
+type KvCredential struct {
+	*store.KvCredential
+	tableName string `gorm:"-"`
+}
+
+// NewKvCredential creates a new in memory static Credential containing a
+// key value secret that is assigned to storeId. Name and description are the only
+// valid options. All other options are ignored.
+func NewKvCredential(
+	ctx context.Context,
+	storeId string,
+	secrets credential.KvSecret,
+	opt ...Option,
+) (*KvCredential, error) {
+	const op = "static.NewKvCredential"
+	var jsonSecrets []byte
+	var err error
+
+	// since the secret is an unordered map of dynamically typed values, the hmac value will not be consistent.
+	// In order to calculate a consistent hmac value, the secret must have a ordered map
+	// which is done by marshalling the secret.
```

Comment:
Nit picking:
```suggestion
	// Since the secret is an unordered map of dynamically typed values, the hmac value will not be consistent.
	// In order to calculate a consistent hmac value, the input must be deterministic,
	// which is done by marshalling the secret.
```